crates/types/src/config/http.rs

```diff
@@ -74,13 +67,21 @@ impl Default for HttpOptions {
         Self {
             http_keep_alive_options: Http2KeepAliveOptions::default(),
             http_proxy: None,
-            no_proxy: Vec::new(),
+            no_proxy: None,
             connect_timeout: NonZeroFriendlyDuration::from_secs_unchecked(10),
             initial_max_send_streams: None,
         }
     }
 }
 
+/// NO_PROXY can be provided as either a comma-separated string `example.com,::1,localhost`, or a list of strings `["example.com", "::1", "localhost"]`
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum NoProxy {
+    List(Vec<String>),
```

Comment:
Maybe add a comment that we have this variant to support the old style of specifying the no proxy option (< v1.6).